spec/controllers/admin/products_controller_spec.rb

```diff
@@ -0,0 +1,148 @@
+require "rails_helper"
+
+RSpec.describe Admin::ProductsController, type: :controller do
+  let(:product) {FactoryBot.create :product}
+  let(:admin){FactoryBot.create :admin}
+  let(:not_admin){FactoryBot.create :user}
+  let(:invalid_params){{name: ""}}
+
+  describe "before action" do
+    before{sign_in admin}
+
+    it {is_expected.to use_before_action :find_product}
+    it {is_expected.to use_before_action :sub_cat}
+    it {is_expected.to use_before_action :set_locale}
+  end
+
+  describe "GET #index" do
```

Comment:
phân biệt describe, it, context

Reply:
Dạ,
describe và context đều là example group
describe: nhóm các test có cùng chức năng hoặc cùng 1 action
context: nhóm các test có cùng chức năng, điều kiện và theo từng trường hợp cụ thể
it dùng để mô tả từng test key